types/state/index.go

```diff
@@ -39,3 +39,31 @@ func (s *StateDB) GetIndex(index Index, chain []byte, key interface{}) ([]byte,
 	s.logInfo("GetIndex", "index", string(index), "chain", hex.EncodeToString(chain), "key", key, "computed", hex.EncodeToString(k[:]))
 	return s.Read(k)
 }
+
+func (tx *DBTransactional) Write(key storage.Key, value []byte) {
```

Comment:
Instead of adding `DBTransactional.Write` and `DBTransactional.WriteIndex` change the receiver of the old method. That ensures that no one can use Write or WriteIndex without a transaction, and it shows what changes if any there are to the method body.